py/src/braintrust/logger.py

```diff
@@ -459,13 +472,20 @@ def __init__(self, log_conn: LazyValue[HTTPConnection]):
 
     def log(self, *args):
         self._start()
+        did_notify_drop = False
         for event in args:
             try:
                 self.queue.put_nowait(event)
             except queue.Full:
                 # Notify consumers to start draining the queue.
                 self.queue_filled_semaphore.release()
-                self.queue.put(event)
+                if not self.queue_drop_when_full:
+                    self.queue.put(event)
+                elif not did_notify_drop:
+                    _logger.warning(
+                        f"Queue is currently full at size {self.queue_maxsize}. Dropping additional elements"
+                    )
+                    did_notify_drop = True
```

Comment:
Small nit: Instead of having this as a boolean I'd have it as a "cooldown" so it's printed at most once every X seconds.

But circling back to Ankur's comments, having some signal about the number of elements dropped is useful as well and I am not sure how you would reconcile only logging once/on cooldown and then logging the dropped count.

Reply:
Yup makes sense! I implemented a batched logging method that logs every N seconds (where N is configurable). Each time we drop records, we add to the counter, and print a log message if it has been more than N seconds since the last one.